server/src/session.ts

```diff
@@ -340,29 +340,25 @@ export class Session {
     }
   }
 
-  private onDefinition(params: lsp.TextDocumentPositionParams) {
-    const {position, textDocument} = params;
-    const filePath = uriToFilePath(textDocument.uri);
-    const scriptInfo = this.projectService.getScriptInfo(filePath);
-    if (!scriptInfo) {
-      this.connection.console.log(`Script info not found for ${filePath}`);
+  private onDefinition(params: lsp.TextDocumentPositionParams): lsp.LocationLink[]|undefined {
+    const lsInfo = this.getLSPositionInfo(params);
+    if (lsInfo === undefined) {
       return;
     }
-
-    const {fileName} = scriptInfo;
-    const langSvc = this.getDefaultLanguageService(scriptInfo);
-    if (!langSvc) {
-      return;
-    }
-
-    const offset = lspPositionToTsPosition(scriptInfo, position);
-    const definition = langSvc.getDefinitionAndBoundSpan(fileName, offset);
+    const {languageService, scriptInfo, offset} = lsInfo;
+    const definition = languageService.getDefinitionAndBoundSpan(scriptInfo.fileName, offset);
     if (!definition || !definition.definitions) {
       return;
     }
     const originSelectionRange = tsTextSpanToLspRange(scriptInfo, definition.textSpan);
+    const results: lsp.LocationLink[] = this.definitionsToLSPLocation(definition.definitions);
+    return results.map(r => ({...r, originSelectionRange}));
+  }
+
+  private definitionsToLSPLocation(definitions: readonly ts.DefinitionInfo[]):
+      Array<{targetUri: string, targetRange: lsp.Range, targetSelectionRange: lsp.Range}> {
```

Comment:
Could we pull the `Array` type parameter out as an interface and add comments about what each are? I realize this is outside the scope of the PR so feel free to disregard, just suggesting because you are already here :)